src/Bicep.Core/LanguageConstants.cs

```diff
@@ -79,7 +79,15 @@ public static class LanguageConstants
 
         public static readonly TypeSymbol Any = new AnyType();
         public static readonly TypeSymbol ResourceRef = CreateResourceScopeReference(ResourceScope.Module | ResourceScope.Resource);
-        public static readonly TypeSymbol ResourceRefArray = new TypedArrayType(ResourceRef, TypeSymbolValidationFlags.Default);
+
+        public static readonly TypeSymbol ResourceOrResourceCollectionRefItem = UnionType.Create(
```

Comment:
Mind adding a comment just to give some context on what this is needed for?

Reply:
Sure...

---
In reply to: [582214962](https://github.com/Azure/bicep/pull/1521#discussion_r582214962) [](ancestors = 582214962)

Reply:
Done.